antarest/study/storage/variantstudy/model/command/update_binding_constraint.py

```diff
@@ -16,7 +25,189 @@
+                matrix_input_paths=[matrix_node_eq.infer_path()],
+                matrix_target_paths=[
+                    matrix_node_lt.infer_target_path(is_link_eq),
+                    matrix_node_gt.infer_target_path(is_link_eq),
+                ],
+            )
+        elif existing_operator == BindingConstraintOperator.LESS:
+            return MatrixInputTargetPaths(
+                matrix_input_paths=[matrix_node_lt.infer_path()],
+                matrix_target_paths=[matrix_node_lt.infer_path(), matrix_node_gt.infer_target_path(is_link_lt)],
+            )
+        elif existing_operator == BindingConstraintOperator.GREATER:
+            return MatrixInputTargetPaths(
+                matrix_input_paths=[matrix_node_gt.infer_path()],
+                matrix_target_paths=[matrix_node_lt.infer_target_path(is_link_gt), matrix_node_gt.infer_path()],
+            )
+        else:
+            raise NotImplementedError(
+                f"Case not handled yet: existing_operator={existing_operator}, new_operator={new_operator}"
+            )
+    elif existing_operator == BindingConstraintOperator.BOTH:
+        if new_operator == BindingConstraintOperator.EQUAL:
+            return MatrixInputTargetPaths(
+                matrix_input_paths=[matrix_node_lt.infer_path(), matrix_node_gt.infer_path()],
+                matrix_target_paths=[matrix_node_eq.infer_target_path(is_link_lt)],
+            )
+        elif new_operator == BindingConstraintOperator.LESS:
+            return MatrixInputTargetPaths(
+                matrix_input_paths=[matrix_node_lt.infer_path(), matrix_node_gt.infer_path()],
+                matrix_target_paths=[matrix_node_lt.infer_target_path(is_link_lt)],
+            )
+        elif new_operator == BindingConstraintOperator.GREATER:
+            return MatrixInputTargetPaths(
+                matrix_input_paths=[matrix_node_lt.infer_path(), matrix_node_gt.infer_path()],
+                matrix_target_paths=[matrix_node_gt.infer_target_path(is_link_gt)],
+            )
+        else:
+            raise NotImplementedError(
+                f"Case not handled yet: existing_operator={existing_operator}, new_operator={new_operator}"
+            )
+    else:
+        raise NotImplementedError(
+            f"Case not handled yet: existing_operator={existing_operator}, new_operator={new_operator}"
+        )
+
+
+def _update_matrices_names(
+    file_study: FileStudy,
+    binding_constraint_id: str,
+    existing_operator: BindingConstraintOperator,
+    new_operator: BindingConstraintOperator,
+) -> None:
+    """
+    Update the matrix file name according to the new operator.
+
+    Args:
+        file_study: the file study
+        binding_constraint_id: the binding constraint ID
+        existing_operator: the existing operator
+        new_operator: the new operator
+
+    Raises:
+        NotImplementedError: if the case is not handled
+    """
+
+    if existing_operator == new_operator:
```

Comment:
Due to legacy matrices generation, we should handle such a case by unlinking every unconcerned matrix.

Reply:
I prefer to do the `unlinking` only when needed. It is the case when you have an operator update.